application_handlers.go

```diff
@@ -55,8 +56,12 @@ func SourceListApplications(c echo.Context) error {
 	applications, count, err = applicationDB.SubCollectionList(m.Source{ID: id}, limit, offset, filters)
 
 	if err != nil {
+		if errors.Is(err, &util.NotFoundError{}) {
```

Comment:
Great! One final little tidbit as far as the comparisons go - we can have a variable over in util that represents this empty error, that way we don't have to rewrite the & and {} over and over again. I'll provide examples at the end of the review.

Reply:
Thanks for comments. My first understanding was when we want to compare error type on this place then we need the variable err contains this new ErrNotFound type, right? And place where this value is created is model/relation_object.go : 127. I changed there

Reply:
Thanks for comments. My first understanding was when we want to compare error type on this place then we need the variable err contains this new ErrNotFound type, right? And place where this value is created is model/relation_object.go : 127. But then I had meeting with Libor and his suggestion was that the creation of concrete error type should be on handler level - so e.g. in application_handlers.go file. But in this case the variable err doesnt contain ErrNotFound type. So second idea is to compare error strings in handlers (because what else can I do?) and then create error type (now only ErrNotFound, but in future ErrBadRequest ....) and probably wrap this in separate function, because this is needed several times in each handler.

Reply:
>  My first understanding was when we want to compare error type on this place then we need the variable err contains this new ErrNotFound type, right? And place where this value is created is model/relation_object.go : 127.

Yes, the `err` would be a `NotFoundError` in [model/relation_object.go#L123](https://github.com/RedHatInsights/sources-api-go/blob/main/model/relation_object.go#L123).

> But then I had meeting with Libor and his suggestion was that the creation of concrete error type should be on handler level - so e.g. in application_handlers.go file. But in this case the variable err doesnt contain ErrNotFound type.

Maybe it's because I come from a different programming language, but personally I would create the concrete error types in the lower levels and then bubble them up to the handlers —or wherever the error must be caught—, and then transform those error types to proper responses.